src/status_im/ui/screens/wallet/manage_connections/views.cljs

```diff
@@ -1,26 +1,42 @@
 (ns status-im.ui.screens.wallet.manage-connections.views
-  (:require [re-frame.core :as re-frame]
+  (:require [quo.core :as quo]
+            [quo.design-system.colors :as colors]
             [quo.react-native :as rn]
+            [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [status-im.ui.components.bottom-panel.views :as bottom-panel]
+            [status-im.ui.components.icons.icons :as icons]
             [status-im.ui.components.react :as react]
+            [status-im.ui.screens.wallet-connect.session-proposal.views :refer [app-management-sheet-view]]
             [status-im.ui.screens.wallet.manage-connections.styles :as styles]
-            [quo.core :as quo]
-            [quo.design-system.colors :as colors]
-            [status-im.ui.components.icons.icons :as icons]
             [status-im.utils.utils :as utils]))
 
-(defn print-session-info [session]
+(defn account-selector-bottom-sheet [session show-account-selector?]
+  (when @show-account-selector?
+    [rn/view {:style {:height 50}}
+     [bottom-panel/animated-bottom-panel
+      session
+      app-management-sheet-view
+      false]]))
+
+(defn print-session-info [session visible-accounts show-account-selector?]
   (let [peer-meta (get-in session [:params 0 :peerMeta])
         peer-id (get-in session [:params 0 :peerId])
         name (get peer-meta :name)
         url (get peer-meta :url)
-        account (get-in session [:params 0  :accounts 0])
+        account (get-in session [:params 0 :accounts 0])
         icons (get peer-meta :icons)
+        dapp-name (get-in session [:params 0 :peerMeta :name])
+        dapp-url (get-in session [:params 0 :peerMeta :url])
         icon-uri (first (status-im.utils.utils/exclude-svg-resources icons))
-        visible-accounts @(re-frame/subscribe [:visible-accounts-without-watch-only])
         selected-account (first (filter
                                  #(= account
                                      (:address %))
-                                 visible-accounts))]
+                                 visible-accounts))
+        _ (re-frame/dispatch [:wallet-connect-legacy/save-session {:peer-id peer-id
```

Comment:
it will be dispatched each time this component is rendered, if you need this you have to use component-did-mount, but even better to move this dispatch in the place where you open this screen

Reply:
I need it to update every time it's opened and updated ..

Reply:
what do you mean by updated ?

Reply:
who updates it ? i mean its not great to dispatch from the view when its updated by subscription , we loose control in that case on that dispatch

Reply:
I mean every time the selector shows or it's updated, then dispatch this event to preserve session data in status-go
But you're right .. That should go to something like [this](https://github.com/status-im/status-mobile/blob/f6eb85edf3a724cfa3aafb545e0e1a2a2e5208cc/src/status_im/ui/screens/wallet/manage_connections/views.cljs#L15) and maybe use component-did-update

Reply:
but still better to move it in the event where we update it, not in the view

Reply:
Please check if it's improved .. Now the event should be executed only happen when it's needed

Reply:
sorry, but we still to move this dispatch to the event handler where we change an account

Reply:
like here `:wallet-connect-legacy/change-session-account`

Reply:
if you need to add it in a few events , not a problem, but it shouldn't be hidden in the component reaction